components/fxa-client/src/http_client.rs

```diff
@@ -103,6 +115,8 @@ pub trait FxAClient {
         client_id: &str,
         scope: &str,
     ) -> Result<HashMap<String, ScopedKeyDataResponse>>;
+    fn fxa_client_configuration(&self, config: &Config) -> Result<ClientConfigurationResponse>;
```

Comment:
I've moved the FxA config fetching in `http_client` so it can be mocked in tests.